dhcpv6/option_clientid.go

```diff
@@ -4,36 +4,31 @@ import (
 	"fmt"
 )
 
-// OptClientId represents a Client ID option
-//
-// This module defines the OptClientId and DUID structures.
-// https://www.ietf.org/rfc/rfc3315.txt
-type OptClientId struct {
-	Cid Duid
+// OptClientID represents a Client Identifier option as defined by RFC 3315
+// Section 22.2.
+func OptClientID(d Duid) Option {
+	return &optClientID{d}
 }
 
-func (op *OptClientId) Code() OptionCode {
-	return OptionClientID
+type optClientID struct {
```

Comment:
do we want an unexported typed with an exported field? And in general I'm not sure this should become private

(same below)

Reply:
I think to the user these types are mostly noise in the godoc. The user wants two things: A way to add the type to options (`OptClientID` function), and a way to get the type from a message/option (`ClientID()` getter)

Where I do think the types should be exported is where it makes multi-member options more readable, e.g.

```go
msg.Options.Update(&OptIANA{
  IAID: id,
  T1: x,
  T2: y,
  Options: IdentityOptions(opts),
})
```
I think is clearer than
```go
msg.Options.Update(OptIANA(id, x, y, opts))
```

Reply:
And yeah, the member should probably be unexported. Though since the Type is unexported, it doesn't make that much of a difference to anything

Reply:
> Though since the Type is unexported, it doesn't make that much of a difference to anything

The thing is that you can return an unexported object, and it can have exported attributes. For example
```
package lib

type unexported struct {
	Name string
}

func Get() *unexported {
	return &unexported{
		Name: "unexported",
	}
}
```

and then calling `lib.Get().Name` from another package will just work.

I remember some linter warning complaining about this type of declaration, but I guess we can live with it